### PR TITLE
Run C++ tests during building

### PR DESCRIPTION
**Pull request by 1e-to**
This PR for testing the running C++ tests in conda recipe

Closes https://github.com/IntelPython/dpctl/issues/98

### CODE REVIEW

File: backends/tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,49 +1,40 @@
...
+find_package(GTest REQUIRED)
+
+# We need thread support for gtest
+find_package(Threads REQUIRED)
+
+set(CMAKE_CTEST_COMMAND ctest -V)
+
+# Emulate autotools like make check target to build tests
+add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
+enable_testing()
```
Comment (PokhodenkoSA) [inline backends/tests/CMakeLists.txt:10]: Please move this line to the top of the file.

File: conda-recipe/build.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ cmake                                                       \
     -DNUMPY_INCLUDE_DIR=${NUMPY_INC}                        \
     ../backends
 
-make -j 4 && make install
+make -j 4 && make check & make install
```
Comment (PokhodenkoSA) [inline conda-recipe/build.sh:35]:
```suggestion
make -j 4 && make check && make install
```

File: conda-recipe/build.sh [hunk 3]
```diff
@@ -32,7 +32,7 @@ cmake                                                       \
     -DNUMPY_INCLUDE_DIR=${NUMPY_INC}                        \
     ../backends
 
-make -j 4 && make install
+make -j 4 && make check && make install
```
Comment (PokhodenkoSA) [inline conda-recipe/build.sh:35]:
```suggestion
make -j 4
make check
make install
```

File: backends/tests/CMakeLists.txt (hunk 1 above)
Comment (PokhodenkoSA) [reply backends/tests/CMakeLists.txt:10]: Not necessary.

**Comment by PokhodenkoSA** [comment]
@1e-to please, make gtest run on Windows too.

**Comment by PokhodenkoSA** [comment]
Strange thing. DPC++ on Windows requires this for access filesystem:
```c++
#include <experimental/filesystem>
namespace fs = std::experimental::filesystem;
```
For Linux it does not.

**Comment by oleksandr-pavlyk** [comment]
@diptorupd added coverage run in public CI which runs C++ tests. 

Is this work still relevant?

**Comment by diptorupd** [comment]
Closing this PR as it is extremely stale and the work to enable gtest in the Teamcity CI will have to be started from scratch.

**Pull request closed without merging**